### PR TITLE
Add a simple example of library use.

### PR DESCRIPTION
**Pull request by derekdreery**
Specifically for internment without any static values.

I thought it would be useful to show a simple use of this library.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/string-cache/204)

### CODE REVIEW

**Comment by derekdreery** [comment]
I'm tidying up my old PRs, I believe this is ready for review/merge.

File: src/lib.rs [hunk 1]
```diff
@@ -7,6 +7,100 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
+//!
+//! A library for interning things that are `AsRef<str>`.
+//!
+//! Some strings may be interned at compile time using the `string-cache-codegen` crate, or the
+//! `EmptyStaticAtomSet` may be used that has no comiple-time interned strings. An `Atom` is an
```
Comment (jdm) [inline src/lib.rs]: sp: comiple

**Comment by derekdreery** [comment]
> These are some great docs; thanks for writing them!

I feel that string interning is something that could be used in places it is not currently, and that it would lead to speed-ups. Since this is the interner used by servo, I assume it is a highly performant one, especially since it allows interning strings at compile-time.

I thought that writing more documentation might lead to more people using it!

**Pull request merged**